### PR TITLE
cpu/saml21: derive low power SRAM length from model number

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

The size of the low power SRAM on SAM L21 family chips varies between members.
Decode the model number just like it's done for STM32 so we don't need to maintain an ever expanding list.

image

### Testing procedure

`tests/periph_backup_ram` should still compile.

### CODE REVIEW

File: cpu/saml21/Makefile.include [hunk 1]
```diff
@@ -13,9 +15,15 @@ endif
...
-  BACKUP_RAM_LEN  = 0x2000
+BACKUP_RAM_ADDR = 0x30000000
+ifeq (18, $(SAM0_DENSITY))
+  BACKUP_RAM_LEN  = 0x2000  # 8k
+else ifeq (17, $(SAM0_DENSITY))
+  BACKUP_RAM_LEN  = 0x2000  # 8k
+else ifeq (16, $(SAM0_DENSITY))
+  BACKUP_RAM_LEN  = 0x1000  # 4k
+else ifeq (15, $(SAM0_DENSITY))
+  BACKUP_RAM_LEN  = 0x8000  # 2k
```
Comment (dylad) [inline cpu/saml21/Makefile.include:26]:
```suggestion
  BACKUP_RAM_LEN  = 0x0800  # 2k
```

**Review by dylad — CHANGES_REQUESTED** [review]
Found one error, otherwise looks good !
Feels free to squash directly.

File: cpu/saml21/Makefile.include (hunk 1 above)
Comment (benpicco) [reply cpu/saml21/Makefile.include]: good catch!

**Review by dylad — APPROVED** [review]
ACK.
`make BOARD=saml21-xpro -C tests/periph_backup_ram` still compiles and `info-build` show the right size of LPSRAM.

**Pull request merged**